src/DocumentDB.ChangeFeedProcessor/FeedProcessing/ChangeFeedQueryTimeoutDecorator.cs

```diff
@@ -0,0 +1,49 @@
+﻿// ----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+// ----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.FeedProcessing
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.DataAccess;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.Monitoring;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.Utils;
+    using Microsoft.Azure.Documents.Client;
+
+    internal class ChangeFeedQueryTimeoutDecorator : IChangeFeedDocumentQuery<Document>
+    {
+        private readonly IChangeFeedDocumentQuery<Document> query;
+        private readonly IHealthMonitor monitor;
+        private readonly TimeSpan timeout;
+
+        public ChangeFeedQueryTimeoutDecorator(IChangeFeedDocumentQuery<Document> query, IHealthMonitor monitor, TimeSpan timeout)
+        {
+            this.query = query;
```

Comment:
we should check input for being null

Reply:
I'll take this back :) we don't do input check for internal, so for consistency we should not have one here.